src/controllers/krate/publish.rs

```diff
@@ -157,6 +157,25 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
         validate_url(documentation.as_deref(), "documentation")?;
         validate_url(repository.as_deref(), "repository")?;
 
+        let keywords = package
+            .keywords
+            .map(|it| it.as_local().unwrap())
+            .unwrap_or_default();
+
+        if keywords.len() > 5 {
+            return Err(cargo_err("expected at most 5 keywords per crate"));
+        }
+
+        for keyword in keywords.iter() {
```

Comment:
Maybe we can extract it as a validation function. The publish function is a bit too long.

Reply:
yeah, that's the plan once I have all of the other things migrated :)

Reply:
Sounds good! :shipit: